tide/master/rest/FileSystemContent.cpp

```diff
@@ -49,27 +49,30 @@
 #include <QUrl>
 
 FileSystemContent::FileSystemContent( const QString& path,
-                                      const MasterConfiguration& config )
+                                      const QString& contentDirectory,
+                                      const QString& contentType )
     : _path ( path )
-    , _config( config )
+    , _contentDirectory ( contentDirectory )
+    , _contentType ( contentType )
 {
 }
 
 std::string FileSystemContent::getTypeName() const
 {
     if( _path == "/" )
-        return "tide/ls";
+        return "tide/"+_contentType.toStdString();
     else
-        return "tide/ls/" + QUrl::fromLocalFile( _path )
-                .path( QUrl::FullyEncoded ).toStdString();
+        return "tide/"+_contentType.toStdString() + "/" +
```

Comment:
if _contentType is always used as an std::string, maybe it could be an std::string all the way, not a QString